### PR TITLE
data schema has merged metadata

### PR DESCRIPTION
**Pull request by mingwli**
createInputMetadata + createOutputMetadata -> createMergedMetadata

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4886)

### CODE REVIEW

**Comment by mingwli** [comment]
image  
Metadata is required as input.

File: src/docfx/build/page/BuildPage.cs [hunk 1]
```diff
@@ -64,52 +67,43 @@ public static async Task<List<Error>> Build(Context context, Document file)
...
                 => new JObject(obj.Properties().OrderBy(p => p.Name));
         }
 
-        private static (List<Error> errors, object output, JObject metadata)
-            CreateDataOutput(Context context, Document file, JObject sourceModel)
-            => (new List<Error>(), context.TemplateEngine.RunJint($"{file.Mime}.json.js", sourceModel), null);
+        private static (object output, JObject metadata)
+            CreateDataOutput(Context context, Document file, JObject sourceModel, JObject mergedMetadata)
+        {
+            sourceModel["metadata"] = mergedMetadata;
```
Comment (OsmondJiang) [inline src/docfx/build/page/BuildPage.cs]: everything will be put into metadata excepted transformedToken?

File: src/docfx/build/metadata/MetadataProvider.cs [hunk 2]
```diff
@@ -41,7 +41,7 @@ public MetadataProvider(Docset docset, Cache cache, MicrosoftGraphCache microsof
             }
         }
 
-        public (List<Error> errors, InputMetadata metadata) GetMetadata(Document file)
+        public (List<Error> errors, InputMetadata metadata) GetInputMetadata(Document file)
```
Comment (live1206) [inline src/docfx/build/metadata/MetadataProvider.cs]: For `MetadataProvider`, should this still be `GetMetadata`, is there any other metadata?

File: src/docfx/build/page/BuildPage.cs (hunk 1 above)
Comment (mingwli) [reply src/docfx/build/page/BuildPage.cs]: Yes according to Adam

File: src/docfx/build/xref/InternalXrefMapBuilder.cs [hunk 3]
```diff
@@ -43,7 +42,7 @@ private static void Load(Context context, ListBuilder<InternalXrefSpec> xrefs, D
                 var callStack = new List<Document> { file };
                 if (file.FilePath.EndsWith(".md", PathUtility.PathComparison))
                 {
-                    var (fileMetaErrors, fileMetadata) = context.MetadataProvider.GetMetadata(file);
+                    var (fileMetaErrors, fileMetadata) = context.MetadataProvider.GetInputMetadata(file);
```
Comment (live1206) [inline src/docfx/build/xref/InternalXrefMapBuilder.cs]:
```suggestion
                    var (fileMetaErrors, fileMetadata) = context.MetadataProvider.GetMetadata(file);
```

File: docs/specs/schema.yml [hunk 4]
```diff
@@ -368,6 +364,33 @@ outputs:
...
+    #YamlMime:TestData
+    prop: value
+    metadata:
+      userMta: user
+  _themes/ContentTemplate/schemas/TestData.schema.json: "{}"
+outputs:
+  a.json: |
+    {
+      "prop": "value",
+      "metadata": {
```
Comment (yufeih) [inline docs/specs/schema.yml]: this makes me wonder if we should merge metadata to `/metadata` for page schema as well.

Comment (OsmondJiang) [reply docs/specs/schema.yml]: We can do that when we are comparing page sdp pipeline?

Comment (mingwli) [reply docs/specs/schema.yml]: yep, i'm wondering too, continue to check with Adam.

**Pull request merged**